### PR TITLE
fetch api for agent users

### PR DESCRIPTION
**Pull request by jvelezpo**
fetches the API to bring agent users data with for current user

### CODE REVIEW

**Comment by gandarez** [comment]
While running `npm i` got this error message, should I skip it or it's a problem?

```
18 verbose stack TypeError: Cannot read property 'match' of undefined
18 verbose stack     at tarballToVersion (/usr/local/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:87:20)
18 verbose stack     at inflatableChild (/usr/local/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:99:22)
18 verbose stack     at /usr/local/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:55:12
18 verbose stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
18 verbose stack     at Object.gotValue (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:155:18)
18 verbose stack     at Object.gotAccum (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:144:25)
18 verbose stack     at Object.tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
18 verbose stack     at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:517:31)
18 verbose stack     at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:574:18)
18 verbose stack     at Promise._settlePromise0 (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:619:10)
18 verbose stack     at Promise._settlePromises (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:699:18)
18 verbose stack     at _drainQueueStep (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:138:12)
18 verbose stack     at _drainQueue (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:131:9)
18 verbose stack     at Async._drainQueues (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:147:5)
18 verbose stack     at Immediate.Async.drainQueues [as _onImmediate] (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:17:14)
18 verbose stack     at processImmediate (internal/timers.js:456:21)
19 verbose cwd /Users/gandarez/Development/git/wakatime/desktop
20 verbose Darwin 19.4.0
21 verbose argv "/usr/local/Cellar/node/13.12.0/bin/node" "/usr/local/bin/npm" "i"
22 verbose node v13.12.0
23 verbose npm  v6.14.4
24 error Cannot read property 'match' of undefined
25 verbose exit [ 1, true ]
```

File: src/components/tray-menu/Tray.tsx [hunk 1]
```diff
@@ -91,6 +92,8 @@ const Tray = ({
     const fetchData = async () => {
       const enabledEditors = await getEditorsState(editorsList);
       ee(enabledEditors);
+
+      const userAgents = ipcRenderer.sendSync('get-user-agents');
```
Comment (gandarez) [inline src/components/tray-menu/Tray.tsx:96]: 'userAgents' is declared but its value is never read

File: src/containers/index.tsx [hunk 2]
```diff
@@ -16,8 +16,10 @@ logger.debug(`isMainProcess ${isMainProcess ? 'true' : 'false'}`);
 if (div) {
   render(
     <Provider store={store}>
-      {/* <InstallEditors />
-    <ActiveEditors /> */}
+      {/*
+        <InstallEditors />
+        <ActiveEditors />
+      */}
```
Comment (gandarez) [inline src/containers/index.tsx:23]: blank space?

File: src/utils/api.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
     try {
-      const apikey = await options.getApiKeyAsync();
-      const token = Buffer.from(apikey).toString('base64');
-      axios.defaults.baseURL = baseURL;
-      axios.defaults.headers.common.Authorization = `Basic ${token}`;
+      await this.setup();
       const response = await axios.get(`/v1/users/current/summaries?start=${start}&end=${end}`);
       if (response.status !== 200) {
-        return null;
+        throw new Error('Get current summaries call failed');
```
Comment (gandarez) [inline src/utils/api.ts:22]: Maybe `failed fetching summaries from api`?

File: src/utils/api.ts [hunk 4]
```diff
@@ -29,4 +35,26 @@ export default class API {
     }
     return summaries.data[0].grand_total.text;
   }
+
+  public async userAgents() {
+    try {
+      await this.setup();
+      const response = await axios.get('/v1/users/current/user_agents');
+      if (response.status !== 200) {
+        throw new Error('Get current user agents call failed');
```
Comment (gandarez) [inline src/utils/api.ts:44]: Maybe `failed fetching user agents from api`?

**Comment by gandarez** [comment]
In general looks good. Don't you mind writting unit tests for the methods I commented above?

**Comment by jvelezpo** [comment]
yes i will do the unit tests in other PR

**Comment by gandarez** [comment]
Would be good to have unit tests for `userAgents` in this PR.

**Comment by jvelezpo** [comment]
here is the PR with testing for API module
https://github.com/wakatime/desktop/pull/108

**Pull request merged**